### DISCUSSION TITLE
Should we use GitHub Discussions?

### DISCUSSION THREAD

**Discussion by ElijahLynn** [department-of-veterans-affairs/va.gov-cms#6832] [General]
Should we use GitHub Discussions and try to have more public discussion (as opposed to Slack) OR turn off the feature?

**Comment by ElijahLynn** [comment]
Well, the Q&A Discussion type wasn't easy to use because you can't post comments, only answers, so I deleted it. This discussion was moved to here from #6831.

![image](https://user-images.githubusercontent.com/1504756/139327297-d8d190d9-f00e-4b81-b41e-44653e89e595.png)

**Reply by ElijahLynn** [reply]
One nice thing is you can actually delete a discussion, as opposed to issues and PRs which aren't deletable, only closeable.

**Comment by ElijahLynn** [comment]
I opened a new Discussion with GitHub "Discussions URLs don't unfurl/expand in Slack" > https://github.com/github/feedback/discussions/6935

**Comment by ElijahLynn** [comment]
Per [US Digital Service Playbook Play #13](https://playbook.cio.gov/#play13) we should be "defaulting to open" so using Discussions vs Slack is an improvement in that direction. The feedback loop is faster with Slack still.

**Comment by chri5tia** [comment]
Personally, I like discussions because it's a more formal approach to document conversations or concepts that can be summarized from the free-flow or stream of consciousness wasteland of slack. I like slack. It's an ad hoc, real time, messaging app that can handle complex or technical discussions. But GitHub "Discussions" can be organized into specific concepts which can highlight slack or other conversations in order to keep more important discussions or topics highlighted.

It would be important to understand the difference between the purpose of slack vs. Discussions vs. the readme.md files.

A workflow as I see it is this:

- Phase 1a: Slack/email/zoom/phone/in person convo
Feedback loop, real time, relevant for specific teams, mostly casual, organized by chronology, technically public but not really, technically archived but not really. Start here, have a conversation, then move the conclusion, summary, consolidated topic to Phase 2

- Phase 1b: code-or-concept-specific iterations tied to PRs and issues, policy drafts, etc.

- Phase 2: Discussions
Organized by topics for all teams, present and future. A good place to find the "conclusion". Not for specific fine-tuned technical things like "Fix this broken link in the readme.md" but more like "Ideal workflow and team collaboration for updating docs?" I feel like a lot of things we use Google Docs and Google sheets and murals could go here or be linked here.

This would also be a great place to discuss specific features and products for VA.gov, so that you can quickly look at the history or concerns of, say, the Find a Form feature. Where teams may work on it in the usual way and summarize it in the VA.gov > Features > Find a Form discussion area. How cool would it be to have links to all the relevant PRs, google docs, slack convos, murals, architectural details, whatever about a *specific* feature or template or custom module.

Other things this would be useful for:

- Policy, cultural norms, bureaucratic necessities and updates, onboarding docs, other "HR" kind of stuff that we have floating around all over and a public forum to air thoughts, praises and grievances about those things.
- A really great place to trickle down the stuff that leadership gets from stakeholders, which they can then link to during their regular "update the team" processes. 

- Phase 3: Forged-in-the-fire 
Phase 1 and 2 lead to final documentation updated in the readme.md and other official docs.

Now for the checks and balances. Adding a new tool such as discussions can be taxing. That workflow should look like this:
1. Did you do a thing or talk about a thing? Was there anything worth summarizing for posterity? Update discussions. OR skip slack and the ticket all together if it's at all substantial and discuss right in Discussions.
2. Did you update a feature or create an issue for a feature? Link it to that feature's Discussions area. 

TL;DR: Forums have gone out of style for some reason, but there really is some value to that type of categorical organization which differs from chronological organization. There is a *lot* of value to keeping the discussions about the code and project and workflow and culture in the same place as the code itself.

**Reply by allaurio8** [reply]
Thanks, I found this very helpful!

**Comment by cvalarida** [comment]
Since this discussion is pretty old now, I'm gonna close it, but the CMS team will revisit the idea later. I've got a few ideas in mind I'll take back to the team. If we decide we want to use GitHub Discussions in some way, we'll write down our plans—probably in another pinned GH Discussion.

Thanks for your input, everybody! 👋
